sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -221,6 +221,9 @@ export class EventReceiver {
         this._receiverOptions.retryOptions,
         abortSignal
       );
+      if (result.length < maxMessageCount) {
+        await this._batchingReceiver.close();
```

Comment:
Add a comment here as to why we are closing the receiver as it wont be clear to the next person